### PR TITLE
#889 Hook up work package stage on creation

### LINKED ISSUE #889: Work Package Stages - Add to the Work Package Creation Flow
### Description

See the epic #345 for background on this feature

### Acceptance Criteria

on work package creation, the user should have the _option_ to select one of the work package stages

### Proposed Solution

- add the dropdown to the wp creation form
- add the field to the wp create route body
- extract the value from teh body in teh controller
- add it to the create function as the last parameter and make it an optional parameter
- make sure the stage gets set in the service function

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Created Select for adding a work package stage + added it to route and service 

## Notes
(its nice that an empty string is sent as null)
## Test Cases
Getting wp when null was passed in as stage
Screen Shot 2023-02-17 at 4 08 30 PM

getting wps when value was passed in
Screen Shot 2023-02-17 at 4 09 27 PM

## Screenshots
Screen Shot 2023-02-17 at 4 05 14 PM

Screen Shot 2023-02-17 at 4 05 31 PM

Screen Shot 2023-02-17 at 4 05 59 PM

Screen Shot 2023-02-17 at 4 05 44 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [ ] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #889  (issue #)

### CODE REVIEW

**Review by RChandler234 — DISMISSED** [review]

File: src/backend/src/routes/work-packages.routes.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ workPackagesRouter.post(
   intMinZero(body('projectWbsNum.carNumber')),
   intMinZero(body('projectWbsNum.projectNumber')),
   intMinZero(body('projectWbsNum.workPackageNumber')),
+  body('stage'),
```
Comment (anthonybernardi) [inline src/backend/src/routes/work-packages.routes.ts:18]: validation?

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 2]
```diff
@@ -148,8 +151,27 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({ a
...
-            <FormControl>
+          <Grid item xs={12} md={6}>
+            <FormControl fullWidth>
+              <FormLabel>Work Package Stage</FormLabel>
+              <Controller
+                name="stage"
+                control={control}
+                render={({ field: { onChange, value } }) => (
+                  <TextField select onChange={onChange} value={value}>
+                    <MenuItem value={''}>None</MenuItem>
```
Comment (anthonybernardi) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:162]: needs a key

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
errors for me when making a work package with no stage (might have to do with the warnings)

also you should merge in the seed data changes and modify that so they have the right stages

File: src/backend/src/routes/work-packages.routes.ts (hunk 1 above)
Comment (Peyton-McKee) [reply src/backend/src/routes/work-packages.routes.ts:18]: what would we validate it as? String or null?

Comment (anthonybernardi) [reply src/backend/src/routes/work-packages.routes.ts]: that it's a valid value in the enum but also optional, look around the other files

Comment (Peyton-McKee) [reply src/backend/src/routes/work-packages.routes.ts]: Do we want to send null from the frontend? Its causing like weird bugs. I dont have the most experience with sending optionals and these weird select control vs uncontrolled components

**Review by anthonybernardi — APPROVED** [review]

**Pull request merged**